webapp/src/pages/eventConfig/components/EventConfigComponent.js

```diff
@@ -22,78 +55,277 @@ class EventConfigComponent extends Component {
       eventService.getEvent(this.props.event.id).then(result => {
         this.setState({
           loading: false,
-          preEvent: result.event,
           updatedEvent: result.event,
-          hasBeenUpdated: false,
           error: result.error
         });
       });
     }
   }
 
-  onClickCancel = () => {
-    this.setState({
-      updatedEvent: this.state.preEvent,
-      hasBeenUpdated: false
-    });
+  onClickCreate = () => {
+    const isValid = this.validateEventDetails();
+    if (isValid) {
+      eventService.create(this.state.updatedEvent).then(result => {
+        this.setState({
+          updatedEvent: result.event,
+          error: result.error
+        });
+        this.props.history.push('/');
```

Comment:
Hmm, it seems that error is returned as 'undefined' even if there is an error. Only statusCode seems to contain anything meaningful - could this be an API/services bug? I have attempted to parse the statusCode, but my current solution feels very hacky, so would welcome comments. In addition, even though I add 'Event key already exists' in errors in setState, the error doesn't actually appear on the page, potentially leaving the user rather confused as to why their event creation failed.

Reply:
Oh interesting! I had a quick look and it seems at first glance to be following the usual pattern which we use on the other pages (and which usually works). Can you use the Network tab to see exactly what JSON gets returned by the backend when there is an error and compare that to what the event service is trying to extract? (https://github.com/deep-learning-indaba/Baobab/blob/a3c7ecd7357afd3ca715150913a439beaab766bb/webapp/src/services/events/events.service.js#L69)  I can have a closer look tonight

Reply:
So I had a closer look and found the issue - it's a bug in the backend. In the [EventAPI](https://github.com/deep-learning-indaba/Baobab/blob/a3c7ecd7357afd3ca715150913a439beaab766bb/api/app/events/api.py#L162) you'll see that all the methods get, post and put are decorated with `@marshal_with(event_fields)`, unfortunately and annoyingly, what this does is try to marshal *all* responses from the method with event_fields, whether they are errors or not (we actually have this bug in quite a few of our APIs :( ). What you can do is remove the `marshal_with` decorators and instead replace the last lines where it has something like `return event, 201` with: `return marshal(event, event_fields), 200`   (see an example in another API [here](https://github.com/deep-learning-indaba/Baobab/blob/a3c7ecd7357afd3ca715150913a439beaab766bb/api/app/attendance/api.py#L130)) - you'll also need to add `marshal` to the imports from `flask_restful` in line 6.

Reply:
Thanks for this clarification. I am not exactly sure what marshal_with decorators are used for, but will investigate further and report back here

Reply:
Doh! I forgot to say what marshal with does haha sorry! 🤦‍♂️  Our rest APIs need to return JSON but you'll see that we actually return an event object in the event API. So all the marshalling does is convert that object to JSON using the mapping between object properties and JSON keys specified in event_fields. The marshal_with decorator just says "convert anything returned by this function to JSON using this mapping". In fact it's not really providing much benefit for us because it's just as much work to construct a dictionary and return that instead, and dictionaries of course map very easily to JSON 🤷‍♂️